app/src/lib/multi-commit-operation.ts

```diff
@@ -0,0 +1,57 @@
+import { Branch } from '../models/branch'
+import {
+  ChooseBranchStep,
+  conflictSteps,
+  MultiCommitOperationStepKind,
+} from '../models/multi-commit-operation'
+import { Popup, PopupType } from '../models/popup'
+import { TipState } from '../models/tip'
+import { IMultiCommitOperationState, IRepositoryState } from './app-state'
+
+/**
+ * Setup the multi commit operation state when the user needs to select a branch as the
+ * base for the operation.
+ */
+export function getMultiCommitOperationChooseBranchStep(
+  state: IRepositoryState,
+  initialBranch?: Branch | null
+) {
+  const {
+    defaultBranch,
+    allBranches,
+    recentBranches,
+    tip,
+  } = state.branchesState
+  let currentBranch: Branch | null = null
+
+  if (tip.kind === TipState.Valid) {
+    currentBranch = tip.branch
+  } else {
+    throw new Error(
+      'Tip is not in a valid state, which is required to start the rebase flow'
+    )
+  }
+
+  const initialState: ChooseBranchStep = {
+    kind: MultiCommitOperationStepKind.ChooseBranch,
+    defaultBranch,
+    currentBranch,
+    allBranches,
+    recentBranches,
+    initialBranch: initialBranch !== null ? initialBranch : undefined,
+  }
+
+  return initialState
```

Comment:
If the return type is explicit as suggested in the comment above, we can now just return the object directly I think:
```suggestion
  return {
    kind: MultiCommitOperationStepKind.ChooseBranch,
    defaultBranch,
    currentBranch,
    allBranches,
    recentBranches,
    initialBranch: initialBranch !== null ? initialBranch : undefined,
  }
```